crates/ide_assists/src/handlers/extract_function.rs

```diff
@@ -112,7 +112,12 @@ pub(crate) fn extract_function(acc: &mut Assists, ctx: &AssistContext) -> Option
 
             let fn_def = format_function(ctx, module, &fun, old_indent, new_indent);
             let insert_offset = insert_after.text_range().end();
-            builder.insert(insert_offset, fn_def);
+            match ctx.config.snippet_cap {
+                Some(_) => {
+                    builder.insert_snippet(ctx.config.snippet_cap.unwrap(), insert_offset, fn_def)
+                }
```

Comment:
```suggestion
            match ctx.config.snippet_cap {
                Some(cap) => builder.insert_snippet(cap, insert_offset, fn_def),
```
Didn't notice that earlier sorry

Reply:
Yeah you're right, didn't notice the anti-pattern, thanks for catching it!